govcd/nsxt_alb_pool_test.go

```diff
@@ -0,0 +1,276 @@
+		Alias:       check.TestName(),
+		Certificate: certificate,
+	}
+	openApiEndpoint, err := getEndpointByVersion(&vcd.client.Client)
+	check.Assert(err, IsNil)
+	createdCertificate, err := adminOrg.AddCertificateToLibrary(certificateConfigWithoutPrivateKey)
+	check.Assert(err, IsNil)
+	PrependToCleanupListOpenApi(createdCertificate.CertificateLibrary.Alias, check.TestName(), openApiEndpoint+createdCertificate.CertificateLibrary.Id)
+
+	poolConfigWithCert := &types.NsxtAlbPool{
+		Name:                   check.TestName() + "-complicated",
+		GatewayRef:             types.OpenApiReference{ID: edgeGatewayId},
+		Algorithm:              "FASTEST_RESPONSE",
+		CaCertificateRefs:      []types.OpenApiReference{types.OpenApiReference{ID: createdCertificate.CertificateLibrary.Id}},
+		CommonNameCheckEnabled: takeBoolPointer(true),
+		DomainNames:            []string{"one", "two", "three"},
+		DefaultPort:            takeIntAddress(1211),
+	}
+
+	testAlbPoolConfig(check, vcd, "CertificateWithNoPrivateKey", poolConfigWithCert, nil, client)
+
+	err = createdCertificate.Delete()
+	check.Assert(err, IsNil)
+}
+
+func testPoolWithCertAndPrivateKey(check *C, vcd *TestVCD, edgeGatewayId string, client *VCDClient) {
+	adminOrg, err := vcd.client.GetAdminOrgByName(vcd.org.Org.Name)
+	check.Assert(err, IsNil)
+	check.Assert(adminOrg, NotNil)
+
+	certificateConfigWithoutPrivateKey := &types.CertificateLibraryItem{
+		Alias:                check.TestName(),
+		Certificate:          certificate,
+		PrivateKey:           privateKey,
+		PrivateKeyPassphrase: "test",
+	}
+
+	openApiEndpoint, err := getEndpointByVersion(&vcd.client.Client)
+	check.Assert(err, IsNil)
+	createdCertificate, err := adminOrg.AddCertificateToLibrary(certificateConfigWithoutPrivateKey)
+	check.Assert(err, IsNil)
+	PrependToCleanupListOpenApi(createdCertificate.CertificateLibrary.Alias, check.TestName(), openApiEndpoint+createdCertificate.CertificateLibrary.Id)
+
+	poolConfigWithCertAndKey := &types.NsxtAlbPool{
+		Name:       check.TestName() + "-complicated",
+		GatewayRef: types.OpenApiReference{ID: edgeGatewayId},
+
+		Algorithm:         "FASTEST_RESPONSE",
+		CaCertificateRefs: []types.OpenApiReference{types.OpenApiReference{ID: createdCertificate.CertificateLibrary.Id}},
+		DefaultPort:       takeIntAddress(1211),
+	}
+
+	testAlbPoolConfig(check, vcd, "CertificateWithPrivateKey", poolConfigWithCertAndKey, nil, client)
+
+	err = createdCertificate.Delete()
+	check.Assert(err, IsNil)
+}
+
+func testAlbPoolConfig(check *C, vcd *TestVCD, name string, setupConfig *types.NsxtAlbPool, updateConfig *types.NsxtAlbPool, client *VCDClient) {
+	fmt.Printf("# Running ALB Pool test with config %s ('System' user: %t) ", name, client.Client.IsSysAdmin)
+
+	edge, err := vcd.nsxtVdc.GetNsxtEdgeGatewayByName(vcd.config.VCD.Nsxt.EdgeGateway)
+	check.Assert(err, IsNil)
+
+	createdPool, err := client.CreateNsxtAlbPool(setupConfig)
+	check.Assert(err, IsNil)
+
```

Comment:
missing check.Assert(createdPool.NsxtAlbPool, NotNil) - in other places too.

Reply:
Fixed